### PR TITLE
check if user is logged in, redirect to project page

### PR DESCRIPTION
**Pull request by sidharthNair**
Checks if the user is already logged in and tries to navigate to login/register page (if so we send them to project page)

### CODE REVIEW

File: src/pages/Login.jsx [hunk 1]
```diff
@@ -1,8 +1,15 @@
 import React from "react";
 import { Form, Input, Button, message } from "antd";
 import { UserOutlined, LockOutlined } from "@ant-design/icons";
-import { post } from "../request";
+import { get, post } from "../request";
+
 export default function Login(props) {
+  get("/is_logged_in").then((response) => {
```
Comment (Ti-tanium) [inline src/pages/Login.jsx:7]: I don't think calling `/is_logged_in` in every page to check credential is a good way. Maybe check at the `App.js`, and do something like in this answer would be better: https://stackoverflow.com/a/43171515/16819261


also, usually, we should not write side effect code (like network calls) in the main body of functional components.

File: src/request.js [hunk 2]
```diff
@@ -10,10 +10,10 @@ const client = axios.create({
 })
 
 export const get = (url) => {
-  return client.get(url)
+  return client.get(url, {withCredentials: true})
```
Comment (Ti-tanium) [inline src/request.js:13]: I didn't see any cookie set up in the backend. Why is this added?

Comment (sidharthNair) [reply src/request.js:13]: In latest PR we use `CORS(app, supports_credentials=True)`, which (from what I understand, I may be wrong about this) should allow the session to persist. From what I've tested on my local machine it seems to work.

File: src/pages/Login.jsx (hunk 1 above)
Comment (sidharthNair) [reply src/pages/Login.jsx:7]: Sounds good, I will try to update this
